classes/05class/gh-actions/README.md

```diff
@@ -0,0 +1,179 @@
+
+## Using Environment Variables
+Environment variables is another feature that can be used in a workflow. There is already a set of default environment variables from GitHub, like `GITHUB_ACTION`, that identifies the id of the action or `GITHUB_SHA`, which is the git commit sha that triggered the worflow and many others, as listed [here](https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables).
+
+It's also possible to create custom environment variables to be used in the workflow, as the example below:
+
+```yaml
+steps:
+  - name: Hello world
+    run: echo Hello world $First_Name $Last_Name!
+    env:
+      First_Name: DevOps
+      Last_Name: Academy
+```
+
+Environment Variables are case sensitive so make sure you have no typos in your workflow file.
+
+## Secrets
+Depending on what a step is executing, it may need a token or a user/password in orther to authenticate. For example, if one of your steps is to publish a docker image in your DockerHub registry, an authentication process will be required so the task can be executed.
+
+For this and other similar cases, we'll need to use the Secrets feature, so we can have this secret encrypted and not in a clear text inside the repository.
+
+A secret can be created inside the `Settings` section of your repository, under the Secrets option. Once you click in the  `Add new secret`, you'll need to give a name and a value.
+<p align="center">
+    <img src="assets/secrets.png">
+</p>
+
+With the secret created you can use it in your workflow. For example, your secret can be used as an Environment Variable on your step, like the example below:
+
+```yaml
+name: Build
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Authenticates
+          mypasssword: ${{ secrets.mypassword }}
+          myusername: ${{ secrets.myusername }}
+        run: docker login -u $myusername -p $mypassword
+      - run: ./build.sh
+        working-directory: ./scripts
+      - run: ./test.sh
+        working-directory: ./scripts
+      - run: ./publish.sh
+        working-directory: ./scripts
+```
+## Workflow Execution
+
+The workflow will be executed based on the event defined on the workflow file. To check the status of the run as well as the logs, you can check on the `Actions` section of your repository. 
+
+The image below shows a set of executions for a workflow and the color on the side of it indicates if it succeds (green), failed (red) or if still in progress (yellow).
+
+<p align="center">
+    <img src="assets/workflows1.png">
+</p>
+
+To get more information about a specific workflow execution, by clicking on it you'll be taken to the workflow logs and you can see all the details and logs for each job/step in your workflow, as the image below:
+<p align="center">
+    <img src="assets/workflows2.png">
+</p>
+
+
```

Comment:
Add link to Marketplace where actions can be found.

Reply:
Added in the next commit.

Reply:
was it added?

Reply:
it should, but probably missed this commit.. it is added now..:)